Add 14.13 release notes (#8622)
chef/chef: RELEASE_NOTES.md

This file holds "in progress" release notes for the current release under development and is intended for consumption by the Chef Documentation team. Please see <https://docs.chef.io/release_notes.html> for the official Chef release notes.

# Chef Client Release Notes 14.13:

## Updated Resources

### directory

> The `directory` has been updated to property set the `deny_rights` permsission on Windows.

Reviewer: ```suggestion
The `directory` has been updated to properly set the `deny_rights` permsission on Windows.
```

### service

The `service` resource is now idempotent on SLES 11 systems.

### cron

The `cron` resource has been updated to advise users to use the specify properties rather than passsing values in as part of the `environment` property. This avoids a situation where a user could pass the differing values in both locations and receive unexpected results.

### link

> The `link` resource includes improved logging on failure to help you debug what has failed.

Reviewer: This could probably use some clarification. Did you mean..:
```suggestion
The `link` resource includes improved logging on failures to help you debug what has failed.
```
and/or
```suggestion
The `link` resource includes improved logging upon failure to help you debug what has failed.
```

### template

The `template` resource now includes additional information when templating failures, which is particularly useful in ChefSpec.

## delete_resource Fix

> The `delete_resource` helper now works properly when the resource you're attempting to delete has multiple providers.

Reviewer: ```suggestion
The `delete_resource` helper now works properly when the resource you are attempting to delete has multiple providers.
```

## Helpers Help Everywhere

> Various helper have been moved into Chef Infra Client's `universal` class, which makes them available anywhere in your cookbook not, not just recipes. If you've ever been confused why something like `search`, `powershell_out`, or `data_bag_item` didn't work somewhere in your code, that should be resolved now.

Reviewer: ```suggestion
Various helpers have been moved into Chef Infra Client's `universal` class, which makes them available anywhere in your cookbook, not just recipes. If you've ever been confused why something like `search`, `powershell_out`, or `data_bag_item` didn't work somewhere in your code, that should be resolved now.
```
